x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/create/helpers.ts

```diff
@@ -275,8 +277,23 @@ export const filterEmptyThreats = (threats: Threats): Threats => {
 };
 
 export const formatDefineStepData = (defineStepData: DefineStepRule): DefineStepRuleJson => {
-  const ruleFields = filterRuleFieldsForType(defineStepData, defineStepData.ruleType);
+  let stepData: Omit<DefineStepRule, 'dataViewId' | 'index' | 'dataSourceType'> & {
+    index?: string[];
+    dataViewId?: string;
+  } = {
+    ...defineStepData,
+  };
+
+  if (defineStepData.dataSourceType === DataSourceType.DataView) {
+    stepData = omit(stepData, 'index');
+  } else if (defineStepData.dataSourceType === DataSourceType.IndexPatterns) {
+    stepData = omit(stepData, 'dataViewId');
+  }
+  stepData = omit(stepData, 'dataSourceType');
```

Comment:
I wonder, is it possible to extract this logic into a separate helper, that would take care about this, and reuse it
Similar code can be found also in [x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx](https://github.com/elastic/kibana/pull/136448/files#diff-14d03cbb6c633190285d6749ffb521827071ea2f0ad920de5a2aa0a52266bc21R657-R662)

Reply:
fixed here: https://github.com/elastic/kibana/pull/136448/commits/a33441eed2364cafea01db4184441d8343b6722a

thanks for the suggestion!